--- origsrc/glib-1.2.10/Makefile.am	2008-07-03 10:44:26.187500000 -0500
+++ src/glib-1.2.10/Makefile.am	2008-07-03 16:01:15.687500000 -0500
@@ -80,6 +80,7 @@
 libglib_la_LDFLAGS = \
 	-version-info $(LT_CURRENT):$(LT_REVISION):$(LT_AGE) \
 	-release $(LT_RELEASE) \
+	-no-undefined \
 	-export-dynamic
 
 noinst_PROGRAMS = testglib testgdate testgdateparser
--- origsrc/glib-1.2.10/gmodule/Makefile.am	2013-07-21 21:03:06.623129000 -0500
+++ src/glib-1.2.10/gmodule/Makefile.am	2013-07-21 21:08:33.432821400 -0500
@@ -36,11 +36,11 @@ libgmodule_la_LIBADD = @G_MODULE_LIBS@ $
 # problem with this ;(
 
 libgplugin_a_la_SOURCES = libgplugin_a.c
-libgplugin_a_la_LDFLAGS = @G_MODULE_LDFLAGS@ -avoid-version -module
-libgplugin_a_la_LIBADD = @G_MODULE_LIBS@ $(libglib)
+libgplugin_a_la_LDFLAGS = @G_MODULE_LDFLAGS@ -avoid-version -module -no-undefined
+libgplugin_a_la_LIBADD = @G_MODULE_LIBS@ $(libglib) libgmodule.la
 
 libgplugin_b_la_SOURCES = libgplugin_b.c
-libgplugin_b_la_LDFLAGS = @G_MODULE_LDFLAGS@ -avoid-version -module
+libgplugin_b_la_LDFLAGS = @G_MODULE_LDFLAGS@ -avoid-version -module -no-undefined
 libgplugin_b_la_LIBADD = @G_MODULE_LIBS@ $(libglib)
 
 noinst_PROGRAMS = testgmodule
