--- origsrc/glib-1.2.10/Makefile.am	2008-07-03 10:44:26.187500000 -0500
+++ src/glib-1.2.10/Makefile.am	2008-07-03 16:01:15.687500000 -0500
@@ -80,6 +80,7 @@
 libglib_la_LDFLAGS = \
 	-version-info $(LT_CURRENT):$(LT_REVISION):$(LT_AGE) \
 	-release $(LT_RELEASE) \
+	-no-undefined \
 	-export-dynamic
 
 noinst_PROGRAMS = testglib testgdate testgdateparser
--- origsrc/glib-1.2.10/gmodule/Makefile.am	2008-07-03 10:44:27.828125000 -0500
+++ src/glib-1.2.10/gmodule/Makefile.am	2008-07-03 10:47:43.578125000 -0500
@@ -26,18 +26,19 @@
 	@G_MODULE_LDFLAGS@ \
 	-version-info $(LT_CURRENT):$(LT_REVISION):$(LT_AGE) \
 	-release $(LT_RELEASE) \
+	-no-undefined \
 	-export-dynamic
 
 libgmodule_la_LIBADD = @G_MODULE_LIBS@  $(libglib)
 
 libgplugin_a_la_SOURCES = libgplugin_a.c
 libgplugin_a_la_DEPENDENCIES = libgmodule.la
-libgplugin_a_la_LDFLAGS = @G_MODULE_LDFLAGS@ -avoid-version -module
+libgplugin_a_la_LDFLAGS = @G_MODULE_LDFLAGS@ -avoid-version -module -no-undefined
 libgplugin_a_la_LIBADD = @G_MODULE_LIBS@  $(libglib) libgmodule.la
 
 libgplugin_b_la_SOURCES = libgplugin_b.c
 libgplugin_b_la_DEPENDENCIES = libgmodule.la
-libgplugin_b_la_LDFLAGS = @G_MODULE_LDFLAGS@ -avoid-version -module
+libgplugin_b_la_LDFLAGS = @G_MODULE_LDFLAGS@ -avoid-version -module -no-undefined
 libgplugin_b_la_LIBADD = @G_MODULE_LIBS@  $(libglib)
 
 noinst_PROGRAMS = testgmodule
--- origsrc/glib-1.2.10/gthread/Makefile.am	2008-07-03 10:44:27.843750000 -0500
+++ src/glib-1.2.10/gthread/Makefile.am	2008-07-03 10:47:49.578125000 -0500
@@ -17,6 +17,7 @@
 libgthread_la_LDFLAGS = \
 	-version-info $(LT_CURRENT):$(LT_REVISION):$(LT_AGE) \
 	-release $(LT_RELEASE) \
+	-no-undefined \
 	-export-dynamic
 
 libgthread_la_LIBADD = @G_THREAD_LIBS@ $(libglib)
